homeassistant/components/isy994/__init__.py

```diff
@@ -134,17 +138,12 @@ async def async_setup_entry(
     hass.data[DOMAIN][entry.entry_id] = {}
     hass_isy_data = hass.data[DOMAIN][entry.entry_id]
 
-    hass_isy_data[ISY994_NODES] = {SENSOR_AUX: [], PROTO_NETWORK_RESOURCE: []}
-    for platform in PLATFORMS:
-        hass_isy_data[ISY994_NODES][platform] = []
-
-    hass_isy_data[ISY994_PROGRAMS] = {}
-    for platform in PROGRAM_PLATFORMS:
-        hass_isy_data[ISY994_PROGRAMS][platform] = []
-
-    hass_isy_data[ISY994_VARIABLES] = {}
-    hass_isy_data[ISY994_VARIABLES][Platform.NUMBER] = []
-    hass_isy_data[ISY994_VARIABLES][Platform.SENSOR] = []
+    hass_isy_data[ISY_NODES] = {p: [] for p in (NODE_PLATFORMS + [SENSOR_AUX])}
+    hass_isy_data[ISY_ROOT_NODES] = {p: [] for p in ROOT_NODE_PLATFORMS}
+    hass_isy_data[ISY_PROGRAMS] = {p: [] for p in PROGRAM_PLATFORMS}
+    hass_isy_data[ISY_VARIABLES] = {p: [] for p in VARIABLE_PLATFORMS}
+    hass_isy_data[ISY_NET_RES] = []
+    hass_isy_data[ISY_DEVICES] = {}
```

Comment:
For a future PR: This might be a bit easier to manage as a `dataclass` instead of a `dict` so you can use named attributes instead

Reply:
@bdraco Good call. I may go ahead and do that before I push the next PR for on level sensors.

Thanks for reviewing all these. I have a few more on deck coming your way while I have time.